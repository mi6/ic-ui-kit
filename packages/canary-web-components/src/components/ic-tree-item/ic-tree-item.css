:host {
  display: block;
  border: none;
  position: relative;
}

:host .tree-item-content {
  display: flex;
  align-items: center;
  height: calc(var(--ic-space-xl) + var(--ic-space-xs));
  padding: 0 var(--ic-space-xs);
  text-decoration: none;
  color: var(--ic-primary-text);

  --ic-typography-color: var(--ic-primary-text);
}

::slotted([slot="router-item"]) {
  display: flex;
  align-items: center;
  height: calc(var(--ic-space-xl) + var(--ic-space-xs));
  text-decoration: none;
  font-family: var(--ic-font-body-family);
  color: var(--ic-primary-text);
  padding: 0 var(--ic-space-xs) !important;
}

::slotted([slot="router-item"].ic-tree-item-single) {
  padding-left: calc(var(--ic-space-xl) + var(--ic-space-xs)) !important;
}

:host .tree-item-content:focus,
::slotted([slot="router-item"]:focus) {
  box-shadow: var(--ic-border-focus);
  border-radius: var(--ic-border-radius);
  transition: var(--ic-transition-duration-fast);
  outline: none;
}

:host .tree-item-content:hover,
::slotted([slot="router-item"]:hover) {
  background-color: var(--ic-action-dark-bg-hover) !important;
  cursor: pointer;
}

:host .tree-item-content:active,
::slotted([slot="router-item"]:active) {
  background-color: var(--ic-action-dark-bg-active) !important;
}

:host(.ic-tree-item-selected) .tree-item-content,
:host(.ic-tree-item-selected) ::slotted([slot="router-item"]) {
  background-color: var(--ic-action-default-bg-hover) !important;
}

:host(.ic-tree-item-small) .tree-item-content,
:host(.ic-tree-item-small) ::slotted([slot="router-item"]) {
  height: var(--ic-space-xl);
}

:host(.ic-tree-item-large) .tree-item-content,
:host(.ic-tree-item-large) ::slotted([slot="router-item"]) {
  height: var(--ic-space-xxl);
}

:host(.ic-tree-item-disabled) {
  pointer-events: none;
}

:host(.ic-tree-item-disabled) .tree-item-content,
:host(.ic-tree-item-disabled) .arrow-dropdown,
:host(.ic-tree-item-disabled) ::slotted([slot="icon"]),
:host(.ic-tree-item-disabled) ::slotted([slot="router-item"]) {
  color: var(--ic-architectural-200) !important;

  --ic-typography-color: var(--ic-architectural-200) !important;

  fill: var(--ic-architectural-200);
  pointer-events: none;
}

:host(.ic-tree-item-light) .tree-item-content,
:host(.ic-tree-item-light) .arrow-dropdown,
:host(.ic-tree-item-light) ::slotted([slot="icon"]),
:host(.ic-tree-item-light) ::slotted([slot="router-item"]) {
  color: var(--ic-architectural-white);

  --ic-typography-color: var(--ic-architectural-white);

  fill: var(--ic-architectural-white);
}

:host .arrow-dropdown {
  color: var(--ic-action-default);
  margin-right: var(--ic-space-xs);
}

:host(.ic-tree-item-small) .arrow-dropdown {
  margin-right: var(--ic-space-xxs);
}

:host .tree-item-expanded {
  transform: rotate(180deg);
  margin-top: calc(-1 * var(--ic-space-xxs));
}

:host .ic-tree-item-single {
  padding-left: calc(var(--ic-space-xl) + var(--ic-space-xs));
}

:host(.ic-tree-item-small) .ic-tree-item-single {
  padding-left: calc(var(--ic-space-xl) + var(--ic-space-xxs));
}

.icon-container {
  width: var(--ic-space-lg);
  height: var(--ic-space-lg);
  margin: 0 var(--ic-space-xs) 0 0;
}

:host(.ic-tree-item-small) .icon-container {
  margin: 0 var(--ic-space-xxs) 0 0;
}

.ic-text-overflow {
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.ic-tooltip-overflow {
  overflow: hidden;
}

/** High Contrast **/
@media (forced-colors: active) {
  ::slotted([slot="icon"]) {
    fill: currentcolor;
  }

  :host .tree-item-content:hover,
  ::slotted([slot="router-item"]:hover),
  :host .tree-item-content:active,
  ::slotted([slot="router-item"]:active),
  :host(.ic-tree-item-selected) .tree-item-content,
  :host(.ic-tree-item-selected) ::slotted([slot="router-item"]) {
    background-color: none !important;
  }

  :host(.ic-tree-item-disabled) .tree-item-content,
  :host(.ic-tree-item-disabled) .arrow-dropdown,
  :host(.ic-tree-item-disabled) ::slotted([slot="icon"]),
  :host(.ic-tree-item-disabled) ::slotted([slot="router-item"]) {
    color: GrayText !important;

    --ic-typography-color: GrayText !important;

    fill: GrayText !important;
  }

  :host .tree-item-content:focus,
  ::slotted([slot="router-item"]:focus) {
    border: var(--ic-hc-border) !important;
  }

  :host(.ic-tree-item-selected) .tree-item-content,
  :host(.ic-tree-item-selected) ::slotted([slot="router-item"]) {
    color: Highlight !important;

    --ic-typography-color: Highlight !important;
  }
}
