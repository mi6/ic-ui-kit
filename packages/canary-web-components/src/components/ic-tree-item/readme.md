# ic-tree-item



<!-- Auto Generated Below -->


## Properties

| Property         | Attribute        | Description                                                                                                                             | Type                                                                                                                                                                                     | Default     |
| ---------------- | ---------------- | --------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `disabled`       | `disabled`       | If `true`, the tree item appears in the disabled state.                                                                                 | `boolean`                                                                                                                                                                                | `false`     |
| `expanded`       | `expanded`       | If `true`, the tree item appears in the expanded state.                                                                                 | `boolean`                                                                                                                                                                                | `false`     |
| `href`           | `href`           | The URL that the tree item link points to. If set, the tree item will render as an "a" tag, otherwise it will render as a div.          | `string`                                                                                                                                                                                 | `undefined` |
| `hreflang`       | `hreflang`       | The human language of the linked URL.                                                                                                   | `string`                                                                                                                                                                                 | `""`        |
| `label`          | `label`          | The label of the tree item.                                                                                                             | `string`                                                                                                                                                                                 | `""`        |
| `referrerpolicy` | `referrerpolicy` | How much of the referrer to send when following the link.                                                                               | `"" \| "no-referrer" \| "no-referrer-when-downgrade" \| "origin" \| "origin-when-cross-origin" \| "same-origin" \| "strict-origin" \| "strict-origin-when-cross-origin" \| "unsafe-url"` | `undefined` |
| `rel`            | `rel`            | The relationship of the linked URL as space-separated link types.                                                                       | `string`                                                                                                                                                                                 | `undefined` |
| `selected`       | `selected`       | If `true`, the tree item appears in the selected state.                                                                                 | `boolean`                                                                                                                                                                                | `false`     |
| `target`         | `target`         | The place to display the linked URL, as the name for a browsing context (a tab, window, or iframe).                                     | `string`                                                                                                                                                                                 | `undefined` |
| `theme`          | `theme`          | Sets the theme color to the dark or light theme color. "inherit" will set the color based on the system settings or ic-theme component. | `"dark" \| "inherit" \| "light"`                                                                                                                                                         | `"inherit"` |


## Events

| Event                | Description                         | Type                           |
| -------------------- | ----------------------------------- | ------------------------------ |
| `icTreeItemSelected` | Emitted when tree item is selected. | `CustomEvent<{ id: string; }>` |


## Methods

### `setFocus() => Promise<void>`

Sets focus on the native `input`.

#### Returns

Type: `Promise<void>`




## Slots

| Slot            | Description                                         |
| --------------- | --------------------------------------------------- |
| `"icon"`        | Content is placed to the left of the label.         |
| `"label"`       | Content is set as the tree item label.              |
| `"router-item"` | Handle routing by nesting your routes in this slot. |


## Dependencies

### Depends on

- ic-tooltip
- ic-typography

### Graph
```mermaid
graph TD;
  ic-tree-item --> ic-tooltip
  ic-tree-item --> ic-typography
  ic-tooltip --> ic-typography
  style ic-tree-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


