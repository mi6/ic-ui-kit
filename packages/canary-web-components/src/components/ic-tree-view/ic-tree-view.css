/**
* @prop --tree-view-width: Width of the tree view
*/

:host {
  display: block;
  width: var(--tree-view-width, 100%);
}

:host .heading-area-container {
  border-bottom: var(--ic-border-default);
  display: flex;
  align-items: center;
  height: calc(var(--ic-space-xl) + var(--ic-space-xs) - var(--ic-space-1px));
  padding: 0 var(--ic-space-xs);
}

:host(.ic-tree-view-small) .heading-area-container {
  height: calc(var(--ic-space-xl) - var(--ic-space-1px));
}

:host(.ic-tree-view-large) .heading-area-container {
  height: calc(var(--ic-space-xxl) - var(--ic-space-1px));
}

:host(.ic-tree-view-light) .heading-area-container {
  color: var(--ic-architectural-white);

  --ic-typography-color: var(--ic-architectural-white);

  fill: var(--ic-architectural-white);
}

.icon-container {
  width: var(--ic-space-lg);
  height: var(--ic-space-lg);
  margin: 0 var(--ic-space-xs) 0 0;
}

:host(.ic-tree-view-small) .icon-container {
  margin: 0 var(--ic-space-xxs) 0 0;
}

.ic-text-overflow {
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.ic-tooltip-overflow {
  overflow: hidden;
}

/** High Contrast **/
@media (forced-colors: active) {
  ::slotted([slot="icon"]) {
    fill: currentcolor;
  }
}
