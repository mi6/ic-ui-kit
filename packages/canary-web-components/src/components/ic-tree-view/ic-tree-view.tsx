import {
  Component,
  h,
  Prop,
  Element,
  Host,
  Watch,
  State,
  Listen,
  forceUpdate,
} from "@stencil/core";
import { IcSizes, IcThemeForegroundNoDefault } from "../../utils/types";
import {
  isPropDefined,
  isSlotUsed,
  checkSlotInChildMutations,
} from "../../utils/helpers";

let treeViewIds = 0;

/**
 * @slot heading - Content is set as the tree view heading.
 * @slot icon - Content is placed to the left of the heading.
 */

@Component({
  tag: "ic-tree-view",
  styleUrl: "ic-tree-view.css",
  shadow: true,
})
export class TreeView {
  private treeViewId = `ic-tree-view-${treeViewIds++}`;
  private treeItemTag = "IC-TREE-ITEM";
  private hostMutationObserver: MutationObserver = null;

  @Element() el: HTMLIcTreeViewElement;

  @State() treeItems: HTMLIcTreeItemElement[];

  /**
   * The appearance of the tree view, e.g. dark, or light.
   */
  @Prop() appearance?: IcThemeForegroundNoDefault = "dark";
  @Watch("appearance")
  watchAppearanceHandler() {
    this.treeItems.forEach((treeItem) => {
      treeItem.appearance = this.appearance;
    });
  }

  /**
   * The heading of the tree view.
   */
  @Prop() heading?: string = "";

  /**
   * The size of the tree view.
   */
  @Prop() size?: IcSizes = "medium";
  @Watch("size")
  watchSizeHandler() {
    this.treeItems.forEach((treeItem) => {
      treeItem.size = this.size;
    });
  }

  disconnectedCallback(): void {
    this.el?.removeEventListener("slotchange", this.setTreeItems);

    this.hostMutationObserver?.disconnect();
  }

  componentDidLoad(): void {
    this.setTreeItems();

    this.watchAppearanceHandler();
    this.watchSizeHandler();
    setTimeout(() => {
      this.truncateTreeViewHeading();
    }, 100);

    this.addSlotChangeListener();

    this.hostMutationObserver = new MutationObserver(this.hostMutationCallback);
    this.hostMutationObserver.observe(this.el, {
      childList: true,
    });
  }

  @Listen("icTreeItemSelected")
  handleTreeItemSelected(event: CustomEvent): void {
    this.treeItems.forEach((treeItem) => {
      if (treeItem.selected && treeItem.id !== event.detail.id) {
        treeItem.selected = false;
      }
    });
  }

  private handleKeyDown = (event: KeyboardEvent): void => {
    const focussedChild = this.treeItems.indexOf(
      this.treeItems.filter((el) => el === document.activeElement)[0]
    );
    const expanded = this.treeItems[focussedChild]?.expanded;
    switch (event.key) {
      case "ArrowDown":
        this.treeItems[
          this.getNextItemToSelect(focussedChild, true)
        ].setFocus();
        if (focussedChild !== this.treeItems.length - 1) {
          event.preventDefault();
        }
        break;
      case "ArrowUp":
        this.treeItems[
          this.getNextItemToSelect(focussedChild, false)
        ].setFocus();
        if (focussedChild !== 0) {
          event.preventDefault();
        }
        break;
      case "ArrowRight":
        if (this.treeItems[focussedChild].isParent && !expanded) {
          this.treeItems[focussedChild].expanded = true;
          this.treeItems[focussedChild].hasParentExpanded = true;
          this.treeItems[focussedChild].updateAriaLabel();
        } else if (this.treeItems[focussedChild].isParent && expanded) {
          (
            this.treeItems[focussedChild].children[0] as HTMLIcTreeItemElement
          ).setFocus();
        }
        event.preventDefault();
        break;
      case "ArrowLeft":
        if (this.treeItems[focussedChild].isParent && expanded) {
          this.treeItems[focussedChild].expanded = false;
          this.treeItems[focussedChild].hasParentExpanded = false;
          this.treeItems[focussedChild].updateAriaLabel();
        } else if (
          this.treeItems[focussedChild].parentElement.tagName ===
          this.treeItemTag
        ) {
          (
            this.treeItems[focussedChild].parentElement as HTMLIcTreeItemElement
          ).setFocus();
        }
        event.preventDefault();
        break;
    }
  };

  private getNextItemToSelect = (
    currentItem: number,
    movingDown: boolean
  ): number => {
    const numItems = this.treeItems.length - 1;

    if (currentItem < 1) {
      currentItem = 0;
    }

    let nextItem = movingDown ? currentItem + 1 : currentItem - 1;
    if (nextItem < 0) {
      nextItem = 0;
    } else if (nextItem > numItems) {
      nextItem = numItems;
    }

    const maxAttempts = numItems + 1;
    let attempts = 0;

    while (attempts < maxAttempts) {
      if (nextItem < 0 || nextItem > numItems) {
        return currentItem;
      }

      if (
        !this.treeItems[nextItem].disabled &&
        (this.treeItems[nextItem].parentElement.tagName !== this.treeItemTag ||
          (this.treeItems[nextItem].parentElement as HTMLIcTreeItemElement)
            .expanded)
      ) {
        return nextItem;
      }

      if (nextItem === numItems && this.treeItems[nextItem].disabled) {
        return currentItem;
      }

      nextItem = movingDown ? nextItem + 1 : nextItem - 1;
      attempts++;
    }

    return currentItem;
  };

  private linkTreeItems = () => {
    this.treeItems.forEach((treeItem) => {
      treeItem.setAttribute("context-id", this.treeViewId);
    });
  };

  private setTreeItems = () => {
    this.treeItems = this.getAllTreeItems(this.el as HTMLElement);

    this.linkTreeItems();
  };

  private getAllTreeItems(element: HTMLElement): HTMLIcTreeItemElement[] {
    const treeItems: HTMLIcTreeItemElement[] = [];

    const collectTreeItems = (el: HTMLElement) => {
      Array.from(el.children).forEach((child) => {
        if (child.tagName === this.treeItemTag) {
          treeItems.push(child as HTMLIcTreeItemElement);
        }

        collectTreeItems(child as HTMLElement);
      });
    };

    collectTreeItems(element);
    return treeItems;
  }

  private addSlotChangeListener = () => {
    this.el.addEventListener("slotchange", this.setTreeItems);
  };

  private truncateTreeViewHeading = () => {
    const typographyEl: HTMLIcTypographyElement =
      this.el.shadowRoot.querySelector(".tree-view-header");
    const tooltip = typographyEl.closest("ic-tooltip");
    const headingContainer: HTMLElement = this.el.shadowRoot.querySelector(
      ".heading-area-container"
    );

    if (typographyEl?.scrollHeight > headingContainer?.clientHeight) {
      typographyEl.classList.add("ic-text-overflow");

      if (!tooltip) {
        const tooltipEl = document.createElement("ic-tooltip");
        tooltipEl.setAttribute("target", this.el.id);
        tooltipEl.setAttribute("label", typographyEl.textContent);
        tooltipEl.classList.add("ic-tooltip-overflow");
        tooltipEl.setAttribute("placement", "right");
        headingContainer.appendChild(tooltipEl);
        tooltipEl.appendChild(typographyEl);
      }
    }
  };

  private hostMutationCallback = (mutationList: MutationRecord[]): void => {
    if (
      mutationList.some(({ type, addedNodes, removedNodes }) =>
        type === "childList"
          ? checkSlotInChildMutations(addedNodes, removedNodes, "icon")
          : false
      )
    ) {
      forceUpdate(this);
    }
  };

  private isHeadingDefined = () =>
    isPropDefined(this.heading) && this.heading !== null;

  private hasHeadingAreaContent = (): boolean => {
    return (
      isSlotUsed(this.el, "heading") ||
      this.isHeadingDefined() ||
      isSlotUsed(this.el, "icon")
    );
  };

  render() {
    const { appearance, heading, size } = this;

    return (
      <Host
        context-id={this.treeViewId}
        class={{
          [`ic-tree-view-${appearance}`]: true,
          [`ic-tree-view-${size}`]: size !== "medium",
        }}
        onKeyDown={this.handleKeyDown}
        aria-label={this.isHeadingDefined() ? heading : null}
      >
        {this.hasHeadingAreaContent() && (
          <div class="heading-area-container">
            {isSlotUsed(this.el, "icon") && (
              <div class="icon-container">
                <slot name="icon" />
              </div>
            )}
            <ic-typography variant="subtitle-large" class="tree-view-header">
              {isSlotUsed(this.el, "heading") ? (
                <slot name="heading" />
              ) : (
                heading
              )}
            </ic-typography>
          </div>
        )}
        <slot></slot>
      </Host>
    );
  }
}
