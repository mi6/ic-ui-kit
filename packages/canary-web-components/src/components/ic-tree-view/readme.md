# ic-tree-view



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute    | Description                                           | Type                             | Default    |
| ------------ | ------------ | ----------------------------------------------------- | -------------------------------- | ---------- |
| `appearance` | `appearance` | The appearance of the tree view, e.g. dark, or light. | `"dark" \| "light"`              | `"dark"`   |
| `heading`    | `heading`    | The heading of the tree view.                         | `string`                         | `""`       |
| `size`       | `size`       | The size of the tree view.                            | `"large" \| "medium" \| "small"` | `"medium"` |


## Slots

| Slot        | Description                                   |
| ----------- | --------------------------------------------- |
| `"heading"` | Content is set as the tree view heading.      |
| `"icon"`    | Content is placed to the left of the heading. |


## CSS Custom Properties

| Name                | Description            |
| ------------------- | ---------------------- |
| `--tree-view-width` | Width of the tree view |


## Dependencies

### Depends on

- ic-tooltip
- ic-typography

### Graph
```mermaid
graph TD;
  ic-tree-view --> ic-tooltip
  ic-tree-view --> ic-typography
  ic-tooltip --> ic-typography
  style ic-tree-view fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


