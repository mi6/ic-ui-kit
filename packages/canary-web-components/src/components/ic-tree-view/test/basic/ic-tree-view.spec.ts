import { newSpecPage } from "@stencil/core/testing";

import { Typography } from "@ukic/web-components/src/components/ic-typography/ic-typography";
import { TreeView } from "../../ic-tree-view";
import { TreeItem } from "../../../ic-tree-item/ic-tree-item";

describe("ic-tree-view component", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [TreeView],
      html: `<ic-tree-view heading="Heading"></ic-tree-view>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with icon", async () => {
    const page = await newSpecPage({
      components: [TreeView],
      html: `<ic-tree-view heading="Heading">
         <svg
            slot="icon"
            xmlns="http://www.w3.org/2000/svg"
            height="24px"
            viewBox="0 0 24 24"
            width="24px"
            fill="#000000"
          >
            <path d="M0 0h24v24H0V0z" fill="none" />
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
      </ic-tree-view>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with tree items", async () => {
    const page = await newSpecPage({
      components: [TreeView, TreeItem],
      html: `<ic-tree-view heading="Heading">
          <ic-tree-item label="Item 1"></ic-tree-item>
          <ic-tree-item label="Item 2"></ic-tree-item>
      </ic-tree-view>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should render with slotted heading", async () => {
    const page = await newSpecPage({
      components: [TreeView, Typography],
      html: `<ic-tree-view>
        <ic-typography slot="heading" variant="subtitle-large">Heading</ic-typography>
      </ic-tree-view>`,
    });

    expect(page.root).toMatchSnapshot();
  });

  it("should test handleTreeItemSelected", async () => {
    const page = await newSpecPage({
      components: [TreeView, TreeItem],
      html: `<ic-tree-view heading="Heading">
          <ic-tree-item label="Item 1"></ic-tree-item>
          <ic-tree-item label="Item 2"></ic-tree-item>
      </ic-tree-view>`,
    });

    const treeItem: HTMLIcTreeItemElement = page.root.querySelector(
      "ic-tree-item#ic-tree-item-2"
    );
    treeItem.selected = true;
    await page.waitForChanges();

    page.rootInstance.handleTreeItemSelected({
      detail: { id: "ic-tree-item-3" },
    });

    expect(treeItem.selected).toBeFalsy();
  });

  it.skip("should test rendering icon slot after initial render", async () => {
    const page = await newSpecPage({
      components: [TreeView],
      html: `<ic-tree-view heading="Heading"></ic-tree-view>`,
    });

    const icon = document.createElement("svg");
    icon.setAttribute("slot", "icon");

    page.rootInstance.hostMutationCallback([
      {
        type: "childList",
        addedNodes: [icon],
        removedNodes: [],
      },
    ]);
  });
});
