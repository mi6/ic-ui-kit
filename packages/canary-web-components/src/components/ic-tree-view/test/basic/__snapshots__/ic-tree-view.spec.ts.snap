// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-tree-view component should render 1`] = `
<ic-tree-view aria-label="Heading" class="ic-tree-view-dark" context-id="ic-tree-view-0" heading="Heading">
  <mock:shadow-root>
    <div class="heading-area-container">
      <ic-typography class="tree-view-header" variant="subtitle-large">
        Heading
      </ic-typography>
    </div>
    <slot></slot>
  </mock:shadow-root>
</ic-tree-view>
`;

exports[`ic-tree-view component should render with icon 1`] = `
<ic-tree-view aria-label="Heading" class="ic-tree-view-dark" context-id="ic-tree-view-1" heading="Heading">
  <mock:shadow-root>
    <div class="heading-area-container">
      <div class="icon-container">
        <slot name="icon"></slot>
      </div>
      <ic-typography class="tree-view-header" variant="subtitle-large">
        Heading
      </ic-typography>
    </div>
    <slot></slot>
  </mock:shadow-root>
  <svg fill="#000000" height="24px" slot="icon" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
    <path d="M0 0h24v24H0V0z" fill="none"></path>
    <path d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"></path>
  </svg>
</ic-tree-view>
`;

exports[`ic-tree-view component should render with slotted heading 1`] = `
<ic-tree-view class="ic-tree-view-dark" context-id="ic-tree-view-3">
  <mock:shadow-root>
    <div class="heading-area-container">
      <ic-typography class="ic-typography-subtitle-large tree-view-header">
        <mock:shadow-root>
          <slot></slot>
        </mock:shadow-root>
        <slot name="heading"></slot>
      </ic-typography>
    </div>
    <slot></slot>
  </mock:shadow-root>
  <ic-typography class="ic-typography-subtitle-large" slot="heading" variant="subtitle-large">
    <mock:shadow-root>
      <slot></slot>
    </mock:shadow-root>
    Heading
  </ic-typography>
</ic-tree-view>
`;

exports[`ic-tree-view component should render with tree items 1`] = `
<ic-tree-view aria-label="Heading" class="ic-tree-view-dark" context-id="ic-tree-view-2" heading="Heading">
  <mock:shadow-root>
    <div class="heading-area-container">
      <ic-typography class="tree-view-header" variant="subtitle-large">
        Heading
      </ic-typography>
    </div>
    <slot></slot>
  </mock:shadow-root>
  <ic-tree-item class="ic-tree-item-dark" context-id="ic-tree-view-2" id="ic-tree-item-0" label="Item 1">
    <mock:shadow-root>
      <div aria-disabled="false" aria-live="polite" class="tree-item-content" tabindex="0">
        <ic-typography class="tree-item-label">
          Item 1
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-tree-item>
  <ic-tree-item class="ic-tree-item-dark" context-id="ic-tree-view-2" id="ic-tree-item-1" label="Item 2">
    <mock:shadow-root>
      <div aria-disabled="false" aria-live="polite" class="tree-item-content" tabindex="0">
        <ic-typography class="tree-item-label">
          Item 2
        </ic-typography>
      </div>
    </mock:shadow-root>
  </ic-tree-item>
</ic-tree-view>
`;
